src/providers/wfs/qgswfsfeatureiterator.cpp

```diff
@@ -1238,6 +1249,19 @@ bool QgsWFSFeatureIterator::fetchFeature( QgsFeature &f )
 
     copyFeature( cachedFeature, f );
     geometryToDestinationCrs( f, mTransform );
+
+    // Retrieve the user-visible id from the Spatialite cache database Id
```

Comment:
Spatialite or sqlite?

Reply:
Spatialite. There are 2 caches: a Spatialite one with the feature with its geom and attributes that gets recreated when reloadData() is called, and the new one SQlite only that only caches gmlid, qgisId and dbId.